### ISSUE TITLE
MSYS2/MinGW, GCC 14.2: target "alsoft.fmt-header-only" does not exist

### ISSUE THREAD

**Issue by LigH-de** [kcat/openal-soft#1076]
Probably related to https://github.com/kcat/openal-soft/commit/0c8df35b10077bd82eef9f7502444f96f21bdc67

```
CMake Error: install(EXPORT "OpenAL" ...) includes target "OpenAL" which requires target "alsoft.fmt-header-only" that is not in any export set.
CMake Error in CMakeLists.txt:
  export called with target "OpenAL" which requires target
  "alsoft.fmt-header-only" that is not in any export set.
```

[logs.zip](https://github.com/user-attachments/files/17977847/logs.zip) collected by [M-AB-S](https://github.com/m-ab-s/media-autobuild_suite)

**Comment by kcat** [comment]
Hmm, odd. OpenAL Soft only "links" with `alsoft::fmt-header-only` (an alias of `alsoft.fmt-header-only`) for static builds, which is only an interface target to set an include path for building OpenAL Soft, and marks it as `PRIVATE`. My understanding of CMake is that should only require the target for building, and not be necessary to consumers of the OpenAL lib. I'm not sure what I'm missing to indicate `alsoft.fmt-header-only` is only necessary for building the library, and doesn't install/export anything.

**Comment by zhongfly** [comment]
https://github.com/kcat/openal-soft/commit/5d2c405c4aea0b6dbcfbcfe147b6a17855627acf does not report errors, while https://github.com/kcat/openal-soft/commit/466204437f7d3e9a9e1cadcb6f26dc6e1b19fd12 has a problem. This suggests that the commit causing the error is in this range. It might be https://github.com/kcat/openal-soft/commit/92894d0698b1f582acef3211752df8dec0240c12

**Comment by L4cache** [comment]
It's `git bisect` time! 😉

**Comment by zhongfly** [comment]
> [5d2c405](https://github.com/kcat/openal-soft/commit/5d2c405c4aea0b6dbcfbcfe147b6a17855627acf) does not report errors, while [4662044](https://github.com/kcat/openal-soft/commit/466204437f7d3e9a9e1cadcb6f26dc6e1b19fd12) has this problem. This suggests that the commit causing the error is in this range. It might be [92894d0](https://github.com/kcat/openal-soft/commit/92894d0698b1f582acef3211752df8dec0240c12)

It is indeed https://github.com/kcat/openal-soft/commit/92894d0698b1f582acef3211752df8dec0240c12 that is causing the error. Hope this helps to solve the problem

**Comment by kcat** [comment]
Should be fixed with commit c238f733faa828c9efc9edb6f2dbc6888ef6ce49. There's a lot of warnings about failing to inline functions, but it otherwise builds fine.
